src/main.js

```diff
@@ -3,15 +3,32 @@ import router from './services/router'
 import App from './App'
 import store from './vuex/store'
 import { sync } from 'vuex-router-sync'
-import { loginFromCookie } from './vuex/modules/auth/actions'
+import { loginFromCookie, checkFirstAdmin } from './vuex/modules/auth/actions'
+import { isConnected } from './services/kuzzleWrapper'
+import Promise from 'bluebird'
 
 Vue.config.debug = process.env.NODE_ENV !== 'production'
 
 sync(store, router)
 
-loginFromCookie(store, () => {
-  router.start({
-    store,
-    components: { App }
-  }, 'body')
-})
+isConnected()
+  .then(() => {
+    return loginFromCookie(store)
+  })
+  .then((user) => {
+    if (!user) {
+      return checkFirstAdmin(store)
+    }
+
+    return Promise.resolve()
+  })
+  .then(() => {
+    router.start({
+      store,
+      components: { App }
+    }, 'body')
+  })
+  .catch((err) => {
+    /* TODO: do something with this... Kuzzle is not responding, what we have to do?? */
```

Comment:
Seems good, but we can't do that I think :/
Before `router.start` we have to check everything for redirect the user to the right route, check rights, and check if kuzzle is alive...
But let's talk about this, it's a good subject